### PR TITLE
Build and deploy the product portal and books on CircleCI

### PR DESCRIPTION
**Pull request by linacambridge**
Cribbing mostly from the [Docusaurus docs](https://docusaurus.io/docs/en/publishing#using-circleci-20).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -232,7 +242,30 @@ jobs:
...
+          command: |
+            pushd docs/sync-storage-handbook
+            mdbook build
+            popd
+            pushd docs/synconomicon
+            mdbook build
+            popd
+            git config --global user.name "Application Services"
+            git config --global user.email "application-services@mozilla.com"
+            echo "machine github.com login linacambridge password $SITE_DEPLOY_TOKEN" > ~/.netrc
```
Comment (linacambridge) [inline .circleci/config.yml]: (I saw we already have `$GITHUB_TOKEN` for Carthage releases...is that yours, @eoger? Would you prefer to reuse it?)

Comment (eoger) [reply .circleci/config.yml]: Is there a way to install an SSH key on CircleCI instead of echo-ing to a file? It makes me a bit uncomfortable

**Comment by vladikoff** [comment]
This is exciting! 🍰

**Comment by rfk** [comment]
I'm also curious what the minimal set of permissions is that you can give to the token and have this work, I fear it's going to be depressingly broad (along the lines of "can push to any branch").

**Comment by linacambridge** [comment]
[This](https://circleci.com/docs/2.0/gh-bb-integration/#deployment-keys-and-user-keys) looks exactly like what we want! It seems GitHub lets you add a deploy key with write access for just that repo, which should work perfectly.

**Pull request merged**